scripts/zones/Grand_Palace_of_HuXzoi/mobs/DE_Quasilumin.lua

```diff
@@ -0,0 +1,307 @@
+        mob:setLocalVar("progress", escortProgress.ENROUTE)
+        local point = 1
+        mob:setLocalVar("point", point)
+        mob:pathThrough(escort.path[point], xi.path.flag.WALK)
+    end
+
+    local now = os.time()
+    local expire = mob:getLocalVar("expire")
+    if expire ~= 0 and expire <= now then
+        if progress ~= escortProgress.COMPLETE then
+            mob:showText(mob, ID.text.TIME_EXCEEDED)
+        end
+        mob:setStatus(xi.status.INVISIBLE)
+        DespawnMob(mob:getID())
+        this.closeDoor(mob)
+        return
+    end
+
+    local opened_door = mob:getLocalVar("opened_door")
+    if opened_door ~= 0 then
+        local npc = GetNPCByID(opened_door)
+        if mob:checkDistance(npc) > 15 then
+            npc:setAnimation(xi.animation.CLOSE_DOOR)
+            mob:setLocalVar("opened_door", 0)
+        elseif npc:getAnimation() ~= xi.animation.OPEN_DOOR then
+            npc:setAnimation(xi.animation.OPEN_DOOR)
+        end
+    end
+
+    for _, doorID in ipairs(escort.doors) do
+        local npc = GetNPCByID(doorID)
+        if doorID ~= opened_door and  mob:checkDistance(npc) <= 8 then
+            npc:setAnimation(xi.animation.OPEN_DOOR)
+            mob:setLocalVar("opened_door", doorID)
+        end
+    end
+end
+
+this.onPath = function(mob)
+    local progress = mob:getLocalVar("progress")
+    local escort = mob:getLocalVar("escort")
+    local data = escorts[escort]
+
+    if data ~= nil and this.shouldMove(mob, progress) then
+        local point = mob:getLocalVar("point")
+        if point == #data.path then
+            mob:showText(mob, ID.text.PATROL_COMPLETED)
+            mob:setLocalVar("progress", escortProgress.COMPLETE)
+            mob:setLocalVar("expire", os.time() + 60)
+        elseif progress ~= escortProgress.COMPLETE then
+            point = point + 1
+            mob:setLocalVar("point", point)
+            mob:pathThrough(data.path[point], xi.path.flag.WALK)
+        end
+    end
+end
+
+this.onTrigger = function(player, mob)
+    local progress = mob:getLocalVar("progress")
+    local point = mob:getLocalVar("point")
+    local escort = mob:getLocalVar("escort")
+    local data = escorts[escort]
+
+    if data ~= nil then
+        if progress == escortProgress.ENROUTE then
+            mob:pathThrough(mob:getPos(), xi.path.flag.NONE)
+            mob:showText(mob, ID.text.PATROL_SUSPENDED)
+            mob:setLocalVar("progress", escortProgress.PAUSED)
+        elseif progress == escortProgress.PAUSED then
+            mob:showText(mob, ID.text.RECOMMENCING_PATROL)
+            mob:setLocalVar("progress", escortProgress.ENROUTE)
+            mob:pathThrough(data.path[point], xi.path.flag.WALK)
+        elseif progress == escortProgress.COMPLETE then
+            mob:showText(mob, ID.text.DUTY_COMPLETE)
+            player:messageSpecial(data.direction)
+            -- TODO: display animation and NPC is not pushing update packet to players in range
```

Comment:
Is this important?

Reply:
I _think_ I was the original person who wrote that comment but it was nearly 3 years ago and I honestly don't remember what it is talking about. I thought about deleting it but thought maybe someone else might know haha

Reply:
Worth leaving it then 👍